From fcd41ee3649ed461f2b2df510480148641f895c3 Mon Sep 17 00:00:00 2001
From: Huang Shan <huang.shan@freescale.com>
Date: Mon, 9 Jun 2014 13:45:58 +0800
Subject: [PATCH 69/70] ENGR00317090 Ethernet apk force close if set a wrong
 ip.

It does not have the code to deal with the illegal input arguments.
Add the code to catch and deal the IllegalArgumentException.

Signed-off-by: Huang Shan <huang.shan@freescale.com>
---
 Ethernet/com/fsl/ethernet/EthernetManager.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Ethernet/com/fsl/ethernet/EthernetManager.java b/Ethernet/com/fsl/ethernet/EthernetManager.java
index 59b8d06..26672f2 100644
--- a/Ethernet/com/fsl/ethernet/EthernetManager.java
+++ b/Ethernet/com/fsl/ethernet/EthernetManager.java
@@ -47,6 +47,7 @@ import android.os.Parcel;
 import android.os.SystemProperties;
 import android.content.SharedPreferences;
 import android.content.SharedPreferences.Editor;
+import android.widget.Toast;
 /**
  * Created by B38613 on 9/27/13.
  */
@@ -206,6 +207,9 @@ public class EthernetManager {
                 Log.e(TAG,"Static IP configuration failed: " + re);
             } catch (IllegalStateException e) {
                 Log.e(TAG,"Static IP configuration fialed: " + e);
+            }catch (IllegalArgumentException e) {
+                Log.e(TAG,"Wrong Static IP: " + e);
+                Toast.makeText(mContext, "Illegal address inputed. You can not access the Internet.",Toast.LENGTH_SHORT).show();
             }
             Log.d(TAG, "set ip manually " + info.toString());
             SystemProperties.set("net.dns1", info.getDnsAddr());
-- 
1.8.0

